wagtail/core/blocks/stream_block.py

```diff
@@ -543,6 +543,33 @@ def get_prep_value(self):
 
         return prep_value
 
+    def blocks_by_type(self):
+        blocks = {}
+        for item in self:
+            block_type = item.block_type
+            block_value = item.value
+
+            if block_type in blocks:
+                blocks[block_type].append(block_value)
+            else:
+                blocks[block_type] = [block_value]
+
+        return blocks
+
+    def blocks_of_type(self, block_type):
+        if block_type not in self.stream_block.child_blocks:
+            return []
+
+        return self.blocks_by_type()[block_type]
+
+    def first_block_of_type(self, block_type):
+        if block_type not in self.stream_block.child_blocks:
+            return
+
+        for item in self:
+            if item.block_type == block_type:
+                return item.value
```

Comment:
I think it would be better for these methods to return the block object (i.e. `return item` here) rather than the value. There are some cases where the block object has some additional functionality that we want to keep - for example, if we have a block like

    heading = CharBlock(template='blocks/heading.html')

where the template is `<h1>{{ value }}</h1>`, then calling {% include_block page.body.blocks_by_type.heading.0 %} should include the `<h1>` tags. This works if we return the block object, but not if we only have the string value.

Reply:
Ah ok. Alright, I will do that.